app/src/main/java/fr/free/nrw/commons/explore/depictions/DepictsClient.java

```diff
@@ -0,0 +1,186 @@
+
+    @Inject
+    public DepictsClient(DepictsInterface depictsInterface, MediaInterface mediaInterface) {
+        this.depictsInterface = depictsInterface;
+        this.mediaInterface = mediaInterface;
+    }
+
+    /**
+     * Search for depictions using the search item
+     * @return list of depicted items
+     */
+    public Observable<DepictedItem> searchForDepictions(String query, int limit, int offset) {
+        return depictsInterface.searchForDepicts(
+                query,
+                String.valueOf(limit),
+                Locale.getDefault().getLanguage(),
+                Locale.getDefault().getLanguage(),
+                String.valueOf(offset)
+        )
+                .flatMap(depictSearchResponse ->Observable.fromIterable(depictSearchResponse.getSearch()))
+                .map(DepictedItem::new);
+    }
+
+    /**
+     * Get URL for image using image name
+     * Ex: title = Guion Bluford
+     * Url = https://upload.wikimedia.org/wikipedia/commons/thumb/0/04/Guion_Bluford.jpg/70px-Guion_Bluford.jpg
+     */
+    private String getThumbnailUrl(String title) {
+        String baseUrl = "https://upload.wikimedia.org/wikipedia/commons/thumb/";
+        title = title.replace(" ", "_");
+        String MD5Hash =  getMd5(title);
+        /**
+         * We use 70 pixels as the size of our Thumbnail (as it is the perfect fits our UI)
+         */
+        return baseUrl + MD5Hash.charAt(0) + '/' + MD5Hash.charAt(0) + MD5Hash.charAt(1) + '/' + title + "/70px-" + title;
+    }
+
+    /**
+     * Ex: entityId = Q357458
+     * value returned = Elgin Baylor Night program.jpeg
+     */
+    public Single<String> getP18ForItem(String entityId) {
+        return depictsInterface.getImageForEntity(entityId)
+                .map(commonsFilename -> {
+                    String name;
+                    try {
+                        JsonObject claims = commonsFilename.getAsJsonObject("claims").getAsJsonObject();
+                        JsonObject p18 = claims.get("P18").getAsJsonArray().get(0).getAsJsonObject();
+                        JsonObject mainsnak = p18.get("mainsnak").getAsJsonObject();
+                        JsonObject datavalue = mainsnak.get("datavalue").getAsJsonObject();
+                        JsonPrimitive value = datavalue.get("value").getAsJsonPrimitive();
+                        name = value.toString();
+                        name = name.substring(1, name.length() - 1);
+                    } catch (Exception e) {
+                        name="";
+                    }
+                    if (!name.isEmpty()){
+                        return getThumbnailUrl(name);
+                    } else return NO_DEPICTED_IMAGE;
+                })
+                .singleOrError();
+    }
+
+    /**
+     * @return list of images for a particular depict entity
+     */
+    public Observable<List<Media>> fetchImagesForDepictedItem(String query, int limit, int sroffset) {
```

Comment:
Please check if `mediaInterface.fetchImagesForDepictedItem` is actually required or one of the existing `MediaClient` methods could be reused.

Reply:
I couldn't say